### DISCUSSION TITLE
Epiphan DVI2USB 3.0 on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by MD-cyb3** [PlusToolkit/PlusLib#1183] [Q&A] [answered]
Hi there,

I have an Epiphan DVI2USB 3.0 frame grabber, and I'd like to us it with the PLUS Toolkit on Ubuntu 22.04.
I've found [this page](http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceEpiphan.html), where it says to 
> install the drivers and software from [VGA2USB-LR download](http://www.epiphan.com/products/frame-grabbers/vga2usb-lr/download/). 

The link doesn't work anymore, but I've found the available Ubuntu drivers [here](https://ssl.epiphan.com/downloads/linux/?dir=Ubuntu/20.04/USB) (following [this link](https://www.epiphan.com/support/vga2usb-lr-drivers-documentation/), under the *Software* tab). However, there's only drivers available for Ubuntu 20.04 the latest. 
Do you know if there is a way to use access the Epiphan on Ubuntu 22.04, even if these drivers are not available for that specific Ubuntu version?

Thanks for helping out!

**Comment by Sunderlandkyl** [comment]
Have you tried using the 20.04 drivers on 22.04?

**Reply by adamrankin** [reply]
If this works, please let us know, as I had another group asking about this as well.

**Reply by MD-cyb3** [reply]
Thanks for your quick replies!

It doesn't work out of the box on my machine, as I'm running a new kernel `6.3.0-060300-generic`, so it prints this 
```
Error! The /var/lib/dkms/vga2usb/3.33.0.17/6.3.0-060300-generic/x86_64/dkms.conf for module vga2usb includes a BUILD_EXCLUSIVE directive which does not match this kernel/arch.
The /var/lib/dkms/vga2usb/3.33.0.17/6.3.0-060300-generic/x86_64/dkms.conf for module vga2usb includes a BUILD_EXCLUSIVE directive which does not match this kernel/arch/config.
This indicates that it should not be built.
Skipped.
```
I wasn't sure if it's worth the effort trying to make it run, so I thought asking first whether someone already managed to do so. I'll look into it, might just install the old `5.4.0-91-generic` and see if it works.

**Reply by adamrankin** [reply]
Not that this is a great answer, but you could try another brand of device, which is ultimately what my answer to the other group was.

**Reply by MD-cyb3** [reply]
Yes, that is also what I might end up doing, but I thought spending at least a little time investigating whether it can be run on Ubuntu 22.04 is worth a shot :shrug:

**Reply by MD-cyb3** [reply, marked as answer]
Actually, I think it is not really possible to install the epiphan drivers on Ubuntu 22.04, as it does not support the old kernel 5.4.0 for which the drivers are provided, and it comes with a `vga2usb_bins.o_shipped` file, without the corresponding sources, so it can't be re-compile for a different kernel... I'll probably stick to another frame grabber then :)
